apps/documents/serializers.py

```diff
@@ -0,0 +1,143 @@
+from django.conf import settings
+
+from enumfields.drf import EnumField
+from rest_framework_json_api import serializers
+from rest_framework.utils.serializer_helpers import ReturnList
+
+from apps.utils import get_s3_client
+from .models import Document, DocumentType, FileType, UploadStatus
+
+
+class DocumentCreateSerializer(serializers.ModelSerializer):
+    """
+    Model serializer for documents validation for s3 signing
+    """
+    document_type = EnumField(DocumentType, ints_as_names=True)
+    file_type = EnumField(FileType, ints_as_names=True)
+    upload_status = EnumField(UploadStatus, ints_as_names=True)
+    shipment_id = serializers.CharField(max_length=36)
+
+    class Meta:
+        model = Document
+        fields = ['name', 'description', 'document_type', 'file_type', 'size', 'upload_status', 'shipment_id']
+
+    def s3_sign(self, document=None):   # pylint: disable=too-many-locals
+        data = self.validated_data
+        s3, _ = get_s3_client()     # pylint: disable=invalid-name
+        bucket = settings.S3_BUCKET
+        file_type = data['file_type']
+        file_type_name = file_type.name.lower()
+
+        # Allowed files type list
+        img_types_list = [enum.name.lower() for enum in FileType]
+        img_types_list.pop(img_types_list.index('pdf'))
```

Comment:
`img_types_list` is something that could be a static `set` instead of creating it every time we call s3_sign.